src/python/pants/backend/python/rules/python_test_runner_integration_test.py

```diff
@@ -130,11 +132,15 @@ def run_pytest(self, *, passthrough_args: Optional[str] = None) -> TestResult:
     if passthrough_args:
       args.append(f"--pytest-args='{passthrough_args}'")
     options_bootstrapper = create_options_bootstrapper(args=args)
-    target = PythonTestsAdaptor(address=Address.parse(f"{self.source_root}:target"))
-    test_result = self.request_single_product(TestResult, Params(target, options_bootstrapper))
-    debug_request = self.request_single_product(
-      TestDebugRequest, Params(target, options_bootstrapper),
+    if origin is None:
+      origin = SingleAddress(directory=self.source_root, name="target")
+    v1_target = self.target(f"{self.source_root}:target")
+    adaptor = PythonTestsAdaptor(
+      address=v1_target.address.to_address(), sources=v1_target._sources_field.sources,
     )
```

Comment:
Previously, the engine would automatically populate the `sources` for the `PythonTestsAdaptor`. It doesn't do that anymore and I'm not sure why. But it does still auto-populate other fields like `dependencies` and `compatibility`?

Reply:
`HydratedTarget` (used to?) depend on `HydratedField` instances: `SourcesField` and `BundlesField`: those would capture `Snapshots` for the sources.

Reply:
I think it still does. But for some reason, when wrapping the `PythonTestsAdaptor` in `PythonTestsAdaptorWithOrigin`, we end up with this exception:

```
AttributeError: 'PythonTestsAdaptor' object has no attribute 'sources'
```

I'll add a TODO to figure out why this is happening. It's not worth spending too much time IMO because this is a single test and this works. I'm more curious than anything.